apps/server/src/modules/tool/uc/tool-reference.uc.spec.ts

```diff
@@ -0,0 +1,201 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { createMock, DeepMocked } from '@golevelup/ts-jest';
+import {
+	ContextExternalToolDO,
+	ExternalToolDO,
+	Permission,
+	SchoolExternalToolDO,
+	ToolConfigurationStatus,
+	ToolReference,
+} from '@shared/domain';
+import { contextExternalToolDOFactory, externalToolDOFactory, schoolExternalToolDOFactory } from '@shared/testing';
+import { Action } from '@src/modules/authorization';
+import { ForbiddenException } from '@nestjs/common';
+import { ToolReferenceUc } from './tool-reference.uc';
+import {
+	CommonToolService,
+	ContextExternalToolService,
+	ExternalToolService,
+	SchoolExternalToolService,
+} from '../service';
+import { ToolContextType } from '../interface';
+
+describe('ToolReferenceUc', () => {
+	let module: TestingModule;
+	let uc: ToolReferenceUc;
+
+	let externalToolService: DeepMocked<ExternalToolService>;
+	let schoolExternalToolService: DeepMocked<SchoolExternalToolService>;
+	let contextExternalToolService: DeepMocked<ContextExternalToolService>;
+	let commonToolService: DeepMocked<CommonToolService>;
+
+	beforeAll(async () => {
+		module = await Test.createTestingModule({
+			providers: [
+				ToolReferenceUc,
+				{
+					provide: ExternalToolService,
+					useValue: createMock<ExternalToolService>(),
+				},
+				{
+					provide: SchoolExternalToolService,
+					useValue: createMock<SchoolExternalToolService>(),
+				},
+				{
+					provide: ContextExternalToolService,
+					useValue: createMock<ContextExternalToolService>(),
+				},
+				{
+					provide: CommonToolService,
+					useValue: createMock<CommonToolService>(),
+				},
+			],
+		}).compile();
+
+		uc = module.get(ToolReferenceUc);
+
+		externalToolService = module.get(ExternalToolService);
+		schoolExternalToolService = module.get(SchoolExternalToolService);
+		contextExternalToolService = module.get(ContextExternalToolService);
+		commonToolService = module.get(CommonToolService);
+	});
+
+	afterAll(async () => {
+		await module.close();
+	});
+
+	describe('getToolReferences', () => {
+		describe('when called with a context type and id', () => {
+			const setup = () => {
+				const userId = 'userId';
+
+				const externalTool: ExternalToolDO = externalToolDOFactory.buildWithId();
+				const schoolExternalTool: SchoolExternalToolDO = schoolExternalToolDOFactory.build({ toolId: externalTool.id });
+				const contextExternalTool: ContextExternalToolDO = contextExternalToolDOFactory
+					.withSchoolExternalToolRef('schoolToolId', 'schoolId')
+					.buildWithId();
+
+				const contextType: ToolContextType = ToolContextType.COURSE;
+				const contextId = 'contextId';
+
+				contextExternalToolService.findAllByContext.mockResolvedValue([contextExternalTool]);
```

Comment:
mockResolvedValue -> mockResolvedValueOnce it should be possible on all places.

Reply:
It is. Changed it